src/components/Article.jsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react"
+import { getArticle, dateFormatter } from "../utils"
+
+
+export const Article = () => {
+
+    const [result, setResult] = useState([])
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(false)
+    const [errMsg, setErrMsg] = useState("Something went wrong, please navigate back to the Articles list and try again")
+    
+    const articleId = window.location.href.split('articles/')[1]
+
+    useEffect(() => {
+        setLoading(true)
+        setError(false)
+        getArticle(articleId)
+        .then((result) => {
+            setResult(result)
+            setLoading(false)
+        })
+        .catch((e) => {
+            if (e.message.includes(400)) {
+                setErrMsg("Articles only have integer id numbers, try selecting one from the Articles list to get an example")
+            } else if (e.message.includes(404)) {
+                setErrMsg("There is no article with this id number!  Try selecting from the Articles list")
+            } else {
+                setErrMsg("Something went wrong, please navigate back to the Articles list and try again")
+            }
+            setLoading(false)
+            setError(true)
+        }) 
+    }, [])
```

Comment:
Please make sure that you fetch every time the ID changes. What would you put in the dependency array in your useEffect?

Reply:
yep see now this, have fixed this with wrapping use effect to respond to changes to path